keps/sig-node/4569-cgroup-v1-maintenance-mode/README.md

```diff
@@ -0,0 +1,637 @@
+
+## Motivation
+
+cgroup v2 offers significant improvements over cgroup v1 in terms of feature set, uniformity of interface, and scalability. With major Linux distributions and container runtimes increasingly supporting cgroup v2, Kubernetes aims to encourage the adoption of this newer, more secure, and more performant technology. By transitioning cgroup v1 support to maintenance mode, Kubernetes can ensure stability for existing deployments while simultaneously promoting the gradual adoption of cgroup v2.
+
+### Goals
+
+1. **Feature Freeze**: No new features will be added to the cgroup v1 support code. The existing functionality of cgroup v1 will be considered complete and stable.
+
+2. **e2e Testing**: Maintain a set of e2e tests to ensure ongoing validation of cgroup v1 for the currently supported features
+
+3. **Best-Effort Bug Fixes**:
+   - Address critical security vulnerabilities in cgroup v1 as a top priority.
+   - Significant functionality breakages in cgroup v1 will be evaluated and potentially fixed if a feasible solution exists.
+   - Acknowledge that some bugs, particularly those requiring substantial changes, may not be resolvable given the constraints around maintaining cgroup v1 support.
+
+4. **Migration Support**: Provide clear and practical migration guidance for users using cgroup v1, facilitating a smoother transition to cgroup v2.
+
+5. **Enhancing cgroup v2 Support**: Address all known pending bugs in Kubernetes’ cgroup v2 support to ensure it reaches a level of reliability and functionality that encourages users to transition from cgroup v1.
+
+### Non-Goals
+
+Removing cgroup v1 support immediately.
+
+## Proposal
+
+The proposal outlines a plan to move cgroup v1 support in Kubernetes into maintenance mode, encouraging the community and users to transition to cgroup v2.
+
+### Risks and Mitigations
+
+The primary risk involves potential disruptions for users who migrate to cgroup v2 with incompatible workloads.
+
+Users depending on the following technologies will need to ensure they are using the specified versions or later, which support cgroup v2:
+
+- OpenJDK / HotSpot: jdk8u372, 11.0.16, 15 and later
+- NodeJs 20.3.0 or later
+- IBM Semeru Runtimes: jdk8u345-b01, 11.0.16.0, 17.0.4.0, 18.0.2.0 and later
+- IBM SDK Java Technology Edition Version (IBM Java): 8.0.7.15 and later
+- If users run any third-party monitoring and security agents that depend on the cgroup file system, they need to update the agents to a version that supports cgroup v2.
+
+Mitigations include providing clear documentation on the migration process, offering community support for common issues encountered during migration, and keeping the cgroup v1 support in the maintenance mode for allowing users additional time to switch to cgroup v2 without any major disruptions.
+
+
+## Design Details
```

Comment:
I think you should mention that we are not creating a feature gate for this feature. We discussed this offline but it isn't really spelled out that we are not introducing a feature gate for this.

Reply:
(we're not adding a feature, either!)

Reply:
Updated, thanks.